connector/subscription_manager.go

```diff
@@ -18,11 +18,11 @@ import (
 
 // SubscriptionManager is responsible for subscriptions management.
 type SubscriptionManager interface {
-	// Add adds a subscription for the provided topic.
-	Add(topic string)
+	// Add adds a subscription for the provided topic. Returns true if it was not already present.
```

Comment:
I would suggest to say: Returns true if it doesn't exist and was successfully subscribed.